Userland/Applications/PixelPaint/Tools/MoveTool.cpp

```diff
@@ -188,17 +188,47 @@ void MoveTool::on_keyup(GUI::KeyEvent& event)
 
 void MoveTool::on_second_paint(Layer const* layer, GUI::PaintEvent& event)
 {
-    if (layer != m_layer_being_moved.ptr() || !m_scaling)
-        return;
-
+    VERIFY(m_editor);
     GUI::Painter painter(*m_editor);
     painter.add_clip_rect(event.rect());
-    auto rect_in_editor = m_editor->content_to_frame_rect(m_new_layer_rect).to_rounded<int>();
-    painter.draw_rect(rect_in_editor, Color::Black);
-    if (!m_cached_preview_bitmap.is_null() || !update_cached_preview_bitmap(layer).is_error()) {
+
+    auto content_rect = m_scaling ? m_new_layer_rect : m_editor->active_layer()->relative_rect();
+    auto rect_in_editor = m_editor->content_to_frame_rect(content_rect).to_rounded<int>();
+    if (m_scaling && (!m_cached_preview_bitmap.is_null() || !update_cached_preview_bitmap(layer).is_error())) {
+        Gfx::PainterStateSaver saver(painter);
         painter.add_clip_rect(m_editor->content_rect());
         painter.draw_scaled_bitmap(rect_in_editor, *m_cached_preview_bitmap, m_cached_preview_bitmap->rect(), 1.0f, Gfx::Painter::ScalingMode::BilinearBlend);
     }
+
+    painter.draw_rect_with_thickness(rect_in_editor, Color::Black, 3);
+    painter.draw_rect_with_thickness(rect_in_editor, Color::White, 1);
+
+    auto resize_anchor_rect_set = resize_anchor_rects(rect_in_editor);
+    for (auto const& resize_anchor_rect : resize_anchor_rect_set.rects()) {
+        painter.draw_rect_with_thickness(resize_anchor_rect, Color::Black, 3);
+        painter.draw_rect_with_thickness(resize_anchor_rect, Color::White, 1);
+    }
```

Comment:
This is good! (I really like the two color lines), but I think this should be split out into two commits. This commit currently seems to be doing two separate things (and that's not clear just from the commit title).

Reply:
I've split the drawing of the two color border around the layer into its own commit.